woocommerce-payments.php

```diff
@@ -98,6 +99,52 @@ function wcpay_init() {
 // If you change the priority of this action, you'll need to change it in the wcpay_check_old_jetpack_version function too.
 add_action( 'plugins_loaded', 'wcpay_init', 11 );
 
+/**
+ * Check WC Payments version and run update related actions.
+ */
+function wcpay_check_version() {
+	if ( defined( 'IFRAME_REQUEST' ) ) {
+		return;
+	}
+
+	$version_option_name = 'wcpay_version';
+
+	$version_option  = get_option( $version_option_name );
+	$requires_update = version_compare( get_option( $version_option_name ), WCPAY_VERSION, '<' );
+
+	if ( ! $version_option || $requires_update ) {
+		wcpay_updated();
+		// WC Payments has been installed or updated.
+		do_action( 'wcpay_updated' );
+
+		if ( ! $version_option ) {
+			// WC Payments has been installed.
+			do_action( 'wcpay_newly_installed' );
+		}
+
+		if ( $requires_update ) {
+			// An existing installation of WC Payments has been updated.
+			do_action( 'wcpay_updated_existing' );
+		}
+	}
+
+	// Add the version option if none is found.
+	if ( ! $version_option ) {
+		add_option( $version_option_name, WCPAY_VERSION );
+	}
+}
+
+add_action( 'init', 'wcpay_check_version', 11 );
```

Comment:
Since we're tracking activation, maybe we should actually use [`register_activation_hook`](https://developer.wordpress.org/reference/functions/register_activation_hook/) to avoid some unnecessary calls during `init`?

Reply:
I first thought about using `register_activation_hook`, but it only runs when a user activates the plugin as far as I know.   I wanted to make sure that we insert the timestamp when the plugin gets updated. I think the [method](https://github.com/Automattic/woocommerce-payments/blob/0be8aea7400b5947b7fde3bb739ac8448ea270f9/includes/class-wc-payments.php#L765) you suggested earlier should do the job. I'm going to remove this action.

Reply:
Oh, that's true! That would have been a mistake. 😬 Good catch!